storage/rinchi/src/main/java/org/openscience/cdk/rinchi/RInChIDecomposition.java

```diff
@@ -0,0 +1,340 @@
+/* Copyright (C) 2024 Uli Fechner
+ *
+ * Contact: cdk-devel@lists.sourceforge.net
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public License
+ * as published by the Free Software Foundation; either version 2.1
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+package org.openscience.cdk.rinchi;
+
+import org.openscience.cdk.ReactionRole;
+import org.openscience.cdk.interfaces.IReaction;
+
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+/**
+ * This class decomposes a RInChI into the individual InChIs and auxiliary Information (if available)
+ * of each reaction component.
+ * Roles of individual components (reactant, product, agent) and the reaction direction are returned.
+ * <p>
+ * A RInChI and its associated RAuxInfo can be decomposed into the constituent InChIs and AuxInfo as follows:
+ * </p>
+ * <pre>
+ * RInChIDecomposition rinchiDecomposition = new RInChIDecomposition(rinchi, rauxinfo).decompose();
+ * List&lt;RInChIDecomposition.Component&gt; components = rinchiDecomposition.getComponents();
+ * for (RInChIDecomposition.Components component: components) {
+ *   System.out.println(component.getInchi())
+ *   if (component.hasAuxInfo) {
+ *       System.out.println(component.getAuxInfo())
+ *   }
+ * }
+ * </pre>
+ *
+ * @author Uli Fechner
+ * @cdk.module rinchi
+ * @cdk.githash
+ */
+public final class RInChIDecomposition extends StatusMessagesOutput {
```

Comment:
Can this be non public?

Reply:
I initially thought that decomposing a RInChI into the constituent InChIs and auxinfo might be a useful functionality, but looking at this now I am not so sure anymore.

I made this non-public as suggested.